### PR TITLE
[Task 1855208] forget SP password when signing out

### PR DESCRIPTION
**Pull request by andxu**
[AB#1855208](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1855208)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/MvpUIHelperImpl.java [hunk 1]
```diff
@@ -29,4 +29,10 @@ public void showException(String msg, Exception e) {
     public String loadPasswordFromSecureStore(@NotNull String key) {
         return IdeaSecureStore.getInstance().loadPassword(key);
     }
+
+    @Deprecated
+    @Override
+    public void removePasswordFromSecureStore(@NotNull String key) {
+        IdeaSecureStore.getInstance().forgetPassword(key);
+    }
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/MvpUIHelperImpl.java:37]: I prefer to keep the method name-prefix/name same in line 35/36

File: Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/IdentityAzureManager.java [hunk 2]
```diff
@@ -150,6 +153,10 @@ public Subscription getSubscriptionById(String sid) {
             }
 
         } catch (Throwable e) {
+            if (StringUtils.isNotBlank(authMethodDetails.getClientId()) && authMethodDetails.getAuthType() == AuthType.SERVICE_PRINCIPAL) {
+                MvpUIHelperFactory.getInstance().getMvpUIHelper().forgetPasswordFromSecureStore(
+                    StringUtils.joinWith("|", "account", authMethodDetails.getClientId()));
```
Comment (Flanker32) [inline Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/IdentityAzureManager.java:158]: Shall we extract `StringUtils.joinWith("|", "account", authMethodDetails.getClientId())` to a method ?

**Review by shenqianjin — APPROVED** [review]
LGTM

File: Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/IdentityAzureManager.java (hunk 2 above)
Comment (andxu) [reply Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/IdentityAzureManager.java:158]: this code will be delete in future, handle it in future

**Pull request merged**